File: 104-maximum-depth-of-binary-tree/104-maximum-depth-of-binary-tree.swift
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public var val: Int
 *     public var left: TreeNode?
 *     public var right: TreeNode?
 *     public init() { self.val = 0; self.left = nil; self.right = nil; }
 *     public init(_ val: Int) { self.val = val; self.left = nil; self.right = nil; }
 *     public init(_ val: Int, _ left: TreeNode?, _ right: TreeNode?) {
 *         self.val = val
 *         self.left = left
 *         self.right = right
 *     }
 * }
 */
class Solution {
    func maxDepth(_ root: TreeNode?) -> Int {
         return maxD(node: root)
    }
    
    func maxD(node: TreeNode?) -> Int {
        if node == nil {
            return 0
        }
        
        var left = 0
        var right = 0
        
        if node?.left != nil {
            left = maxD(node: node?.left)
        }
        
        if node?.right != nil {
           right = maxD(node: node?.right)
        }
        
        return  1 + max(left, right)
    }
}

File: 104-maximum-depth-of-binary-tree/104-maximum-depth-of-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    
      def dfs(self, node: TreeNode):
        if not node: 
            return 0 
        
        left = self.dfs(node.left)
        right = self.dfs( node.right)
        
        return  1 + max(left, right)
    
      def maxDepth(self, root: Optional[TreeNode]) -> int:
          if not root:
             return 0 
        
          return 1 + max(self.dfs(root.left), self.dfs(root.right))
            
  
        
       
        